src/modules/cpu/kernel/crop_and_patch.hpp

```diff
@@ -0,0 +1,1540 @@
+            for(int i = 0; i < roi.xywhROI.roiHeight; i++)
+            {
+                Rpp16f *srcPtr1Temp, *srcPtr2Temp, *dstPtrTempR, *dstPtrTempG, *dstPtrTempB;
+                srcPtr2Temp = srcPtr2Row;
+                dstPtrTempR = dstPtrRowR;
+                dstPtrTempG = dstPtrRowG;
+                dstPtrTempB = dstPtrRowB;
+                int vectorLoopCount = 0;
+                if(i >= patchRoi->xywhROI.xy.y && i < (patchRoi->xywhROI.xy.y + cropRoi->xywhROI.roiHeight))
+                {
+#if __AVX2__
+                    for(; vectorLoopCount < patchAlignedLength1; vectorLoopCount += vectorIncrement)
+                    {
+                        Rpp32f srcPtrTemp_ps[12], dstPtrTemp_ps[12];
+                        for(int cnt = 0; cnt < 12; cnt++)
+                            *(srcPtrTemp_ps + cnt) = (Rpp32f) *(srcPtr2Temp + cnt);
+
+                        __m128 p[4];
+                        rpp_simd_load(rpp_load12_f32pkd3_to_f32pln3, srcPtrTemp_ps, p);    // simd loads
+                        rpp_simd_store(rpp_store12_f32pln3_to_f32pln3, dstPtrTemp_ps, dstPtrTemp_ps + 4, dstPtrTemp_ps + 8, p);    // simd stores
+                        for(int cnt = 0; cnt < 4; cnt++)
+                        {
+                            *(dstPtrTempR + cnt) = (Rpp16f) *(dstPtrTemp_ps + cnt);
+                            *(dstPtrTempG + cnt) = (Rpp16f) *(dstPtrTemp_ps + 4 + cnt);
+                            *(dstPtrTempB + cnt) = (Rpp16f) *(dstPtrTemp_ps + 8 + cnt);
+                        }
+                        srcPtr2Temp += vectorIncrement;
+                        dstPtrTempR += vectorIncrementPerChannel;
+                        dstPtrTempG += vectorIncrementPerChannel;
+                        dstPtrTempB += vectorIncrementPerChannel;
+                    }
+#endif
+                    for (; vectorLoopCount < patchBufferLength1; vectorLoopCount += 3)
+                    {
+                        *dstPtrTempR++ = srcPtr2Temp[0];
+                        *dstPtrTempG++ = srcPtr2Temp[1];
+                        *dstPtrTempB++ = srcPtr2Temp[2];
+                        srcPtr2Temp += 3;
+                    }
+                    vectorLoopCount = 0;
+                    srcPtr1Temp = srcPtr1Row;
+#if __AVX2__
+                    for (; vectorLoopCount < cropAlignedLength; vectorLoopCount += vectorIncrement)
+                    {
+                        Rpp32f srcPtrTemp_ps[12], dstPtrTemp_ps[12];
+                        for(int cnt = 0; cnt < 12; cnt++)
+                            *(srcPtrTemp_ps + cnt) = (Rpp32f) *(srcPtr1Temp + cnt);
+
+                        __m128 p[4];
+                        rpp_simd_load(rpp_load12_f32pkd3_to_f32pln3, srcPtrTemp_ps, p);    // simd loads
+                        rpp_simd_store(rpp_store12_f32pln3_to_f32pln3, dstPtrTemp_ps, dstPtrTemp_ps + 4, dstPtrTemp_ps + 8, p);    // simd stores
+                        for(int cnt = 0; cnt < 4; cnt++)
```

Comment:
why can't we store directly to dstPtrTempR, G and B and get rid of this loop here

Reply:
This has been the case only for f16 since pre-AVX512. Shall get back on this on a separate call